example: Start spining after page loaded

rzr opened the pull request:
Example to show to programmatically control properties

[![webthings-webapp-aframe](https://cf.mastohost.com/v1/AUTH_91eb37814936490c95da7b85993cc2ff/socialsamsunginternet/media_attachments/files/000/007/588/original/3be48d764712ab7e.png)](https://social.samsunginter.net/@rzr/101742800300601130# "webthings-webapp-aframe")

Change-Id: I140ec2892d3b2a4524183630984b6c7ee7fb6eb9
Relate-to: https://github.com/aframevr/aframe/blob/master/docs/introduction/javascript-events-dom-apis.md
Signed-off-by: Philippe Coval <p.coval@samsung.com>

rzr committed:
toogle: Update external changes

This change allows an element to be changed programmatically:

```js
valueSlider.setAttribute('ui-slider', 'value', Math.random());
```

Change-Id: I140ec2892d3b2a4524183630984b6c7ee7fb6eb9
Bug: https://github.com/caseyyee/aframe-ui-widgets/pull/17
Relate-to: https://github.com/rzr/webthing-iotjs/wiki/XR
Signed-off-by: Philippe Coval <p.coval@samsung.com>

caseyyee commented on examples/basic/index.html:
```diff
@@ -95,6 +95,9 @@
         clearInterval(interval);
       }
     });
+    setTimeout(function(){
```
Maybe rather than doing a `setTimeout`, we can just set the value to 1 in the markup so that it is spinning from the start?

rzr replied:
Well I can change this to static, but I the reason I prefered to delay the start was mostly to show how to change attribute dynamically from javascript... this is where that update method is relevant... so I thought this example will illustrate more the real use.

For the context I plan to use widgets also in read only mode and change them from external events, eg:
http://github.com/SamsungInternet/color-sensor-js

caseyyee replied:
I see. Just don't want anyone confused about why the value changed if they don't look at the code.  Maybe one possibility is to hook up one of the buttons so that it toggles the switch? A bit of labeling using `<a-text>` would go a long way as well.  See https://aframe.io/docs/0.9.0/primitives/a-text.html

Given that the repo is pretty old, i'm going to bump the a-frame version up to something more current.  Hopefully that won't break anything for you?

rzr replied:
yea this is what i am actually doing with my code ... 

I have tested current aframe it seems to work , but I need to sort out how to use slider with mouse or gearvr2 controller..

caseyyee replied:
Mmm. Yeah, mouse-click and pan is difficult since it's also used for panning/rotating the scene. Any ideas?

caseyyee commented:
Thanks for this :)  I think besides that, the comment made about setTimeout, the rest looks good :)

caseyyee reviewed (changes requested):
Thanks for this :) I think besides that, the comment made about setTimeout, the rest looks good.

rzr committed:
slider: Update external changes

This change allows an element to be changed programmatically:

```js
let value = Number(valueSlider.getAttribute('ui-slider')['max'] || 1);
value *= Math.random();
valueSlider.setAttribute('ui-slider', 'value', value);
```

Change-Id: Ie9c2fa400ffaf2b555cee9d7f81fa9f4c4977d36
Bug: https://github.com/caseyyee/aframe-ui-widgets/pull/17
Relate-to: https://github.com/rzr/webthing-iotjs/wiki/XR
Signed-off-by: Philippe Coval <p.coval@samsung.com>

rzr committed:
build: Regenerate dist files

Change-Id: I6a3f8c788ead485d0ed9d45856f1132b562b6b61
Signed-off-by: Philippe Coval <p.coval@samsung.com>

caseyyee reviewed (approved):
OK, going to accept these changes.

Thanks @rzr!
